### DISCUSSION TITLE
Download error

### DISCUSSION THREAD

**Discussion by CodeScript-vd** [kolbasa/cordova-plugin-apkupdater#34] [Q&A] [answered]
I receive the following error when downloading the file from an ip, can you help me

{
    "message": "Download failed: Cleartext HTTP traffic to xxx.xxx.xxx.xxx not permitted (java.io.IOException)",
    "stack": "de.kolbasa.apkupdater.exceptions.DownloadFailedException: Download failed: Cleartext HTTP traffic to xxx.xxx.xxx.xxx not permitted (java.io.IOException)\n  at com.android.okhttp.HttpHandler$CleartextURLFilter.checkURLPermitted(HttpHandler.java:127)\n  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:462)\n  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:131)\n  at de.kolbasa.apkupdater.downloader.FileDownloader.download(FileDownloader.java:47)\n  at de.kolbasa.apkupdater.update.UpdateManager.downloadFile(UpdateManager.java:76)\n  at de.kolbasa.apkupdater.update.UpdateManager.download(UpdateManager.java:124)\n  at de.kolbasa.apkupdater.ApkUpdater.download(ApkUpdater.java:110)\n  at de.kolbasa.apkupdater.ApkUpdater.lambda$execute$3$ApkUpdater(ApkUpdater.java:243)\n  at de.kolbasa.apkupdater.-$$Lambda$ApkUpdater$i2uPxQeilYT0voSmjrvq6lzNQe0.run(Unknown Source:6)\n  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\n  at java.lang.Thread.run(Thread.java:923)\n"
}

**Comment by kolbasa** [comment]
Plaintext HTTP is not allowed by default, you need to put `android:usesCleartextTraffic="true"` in your `AndroidManifest.xml` in the `application` tag.

With Cordova this can be controlled via `config.xml`:
```xml
<platform name="android">
    <edit-config file="app/src/main/AndroidManifest.xml" mode="merge" target="/manifest/application">
        <application android:usesCleartextTraffic="true" />
    </edit-config>
</platform>
```

**Comment by kolbasa** [comment, marked as answer]
I'm not sure about the latest Ionic versions.
See this forum thread for more details: https://forum.ionicframework.com/t/err-cleartext-not-permitted-in-debug-app-on-android/164101/86

**Reply by CodeScript-vd** [reply]
I already had it added in my config.xml, but it wasn't working I added this in my network_security_config.xml

![Captura de pantalla 2021-08-22 142148](https://user-images.githubusercontent.com/65878998/130367446-68021623-6751-4051-8132-4de5d31ac65c.png)

thank you.
